Tree sitter 0.25 does not compile with no_std ("std" feature disabled)

### Problem

It seems that when bumping dependencies to 0.25 there are references to std in the lib.rs:
```
Compiling tree-sitter v0.25.3
error[E0433]: failed to resolve: use of undeclared crate or module `std`
    --> /home/simon/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.3/binding_rust/lib.rs:1225:38
     |
1225 |             let (c, len) = D::decode(std::slice::from_raw_parts(data, len as usize));
     |                                      ^^^ use of undeclared crate or module `std`
     |
help: consider importing one of these modules
     |
10   + use crate::slice;
     |
10   + use core::slice;
     |
10   + use alloc::slice;
     |
help: if you import `slice`, refer to it directly
     |
1225 -             let (c, len) = D::decode(std::slice::from_raw_parts(data, len as usize));
1225 +             let (c, len) = D::decode(slice::from_raw_parts(data, len as usize));
     |

error[E0433]: failed to resolve: use of undeclared crate or module `std`
    --> /home/simon/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.3/binding_rust/lib.rs:1425:17
     |
1425 |                 std::ptr::from_ref::<AtomicUsize>(flag).cast::<usize>(),
     |                 ^^^ use of undeclared crate or module `std`
     |
help: consider importing this module
     |
10   + use core::ptr;
     |
help: if you import `ptr`, refer to it directly
     |
1425 -                 std::ptr::from_ref::<AtomicUsize>(flag).cast::<usize>(),
1425 +                 ptr::from_ref::<AtomicUsize>(flag).cast::<usize>(),
     |

error[E0599]: no method named `stop_printing_dot_graphs` found for mutable reference `&mut Parser` in the current scope
    --> /home/simon/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.3/binding_rust/lib.rs:1435:14
     |
1435 |         self.stop_printing_dot_graphs();
     |              ^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut Parser`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `tree-sitter` (lib) due to 3 previous errors
```

### Steps to reproduce

Update tree_sitter to 0.25 in Cargo.toml in a no_std project:
```
[dependencies]
tree-sitter = { version = "0.25", default-features = false, features = [] }
```

### Expected behavior

Builds without std

### Tree-sitter version (tree-sitter --version)

0.25.3

### Operating system/version

Arch Linux

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -238,7 +238,6 @@ jobs:
       if: ${{ !matrix.no-run && !matrix.use-cross && inputs.run-test }}
       run: $BUILD_CMD run -p xtask -- test-wasm
 
-
     - name: Upload CLI artifact
       uses: actions/upload-artifact@v4
       with:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -203,6 +203,13 @@ jobs:
         npm run build
         npm run build:debug
 
+    - name: Check no_std builds
+      if: ${{ !matrix.no-run && inputs.run-test }}
+      shell: bash
+      run: |
+        cd lib
+        $BUILD_CMD check --no-default-features
+
     - name: Build target
       run: $BUILD_CMD build --release --target=${{ matrix.target }} --features=${{ matrix.features }}
 
@@ -230,6 +237,7 @@ jobs:
     - name: Run wasm tests
       if: ${{ !matrix.no-run && !matrix.use-cross && inputs.run-test }}
       run: $BUILD_CMD run -p xtask -- test-wasm
+
 
     - name: Upload CLI artifact
       uses: actions/upload-artifact@v4

--- a/lib/binding_rust/ffi.rs
+++ b/lib/binding_rust/ffi.rs
@@ -16,6 +16,7 @@ extern "C" {
 }
 
 #[cfg(windows)]
+#[cfg(feature = "std")]
 extern "C" {
     pub(crate) fn _ts_dup(handle: *mut std::os::raw::c_void) -> std::os::raw::c_int;
 }

--- a/lib/binding_rust/lib.rs
+++ b/lib/binding_rust/lib.rs
@@ -1222,7 +1222,7 @@ impl Parser {
             len: u32,
             code_point: *mut i32,
         ) -> u32 {
-            let (c, len) = D::decode(std::slice::from_raw_parts(data, len as usize));
+            let (c, len) = D::decode(core::slice::from_raw_parts(data, len as usize));
             if let Some(code_point) = code_point.as_mut() {
                 *code_point = c;
             }
@@ -1422,7 +1422,7 @@ impl Parser {
         if let Some(flag) = flag {
             ffi::ts_parser_set_cancellation_flag(
                 self.0.as_ptr(),
-                std::ptr::from_ref::<AtomicUsize>(flag).cast::<usize>(),
+                core::ptr::from_ref::<AtomicUsize>(flag).cast::<usize>(),
             );
         } else {
             ffi::ts_parser_set_cancellation_flag(self.0.as_ptr(), ptr::null());

--- a/lib/binding_rust/lib.rs
+++ b/lib/binding_rust/lib.rs
@@ -1432,7 +1432,10 @@ impl Parser {
 
 impl Drop for Parser {
     fn drop(&mut self) {
-        self.stop_printing_dot_graphs();
+        #[cfg(feature = "std")]
+        {
+            self.stop_printing_dot_graphs();
+        }
         self.set_logger(None);
         unsafe { ffi::ts_parser_delete(self.0.as_ptr()) }
     }

